### ISSUE TITLE
False positive in redundantFileprivate rule when using type inference with initializers

### ISSUE THREAD

**Issue by msanders** [nicklockwood/SwiftFormat#540]
Hi, I ran into this today when running `swiftformat` on a codebase of mine:

```swift
struct Example {
    fileprivate init() {}
}

enum Namespace {
    static let example: Example = .init()
}
```

#### Expected behavior

This is a valid use of `fileprivate`, so it shouldn't automatically change to `private` (which will cause a compiler error) or throw a lint error.

#### Actual behavior

It shows the following warning:

```sh
$ swiftformat --version
0.44.0
$ swiftformat --lint ./example.swift
Running SwiftFormat...
(lint mode - no files will be changed.)
/tmp/swiftformat/example.swift:2:1: warning: (redundantFileprivate) Prefer private over fileprivate where equivalent.
SwiftFormat completed in 0.02s.
Source input did not pass lint check.
1/1 files require formatting.
```

Then, when running the formatter it causes this error:

```sh
$ swiftformat ./example.swift                                                                                                           
Running SwiftFormat...
SwiftFormat completed in 0.02s.
1/1 files formatted.
$ swift ./example.swift
./example.swift:6:36: error: 'Example' initializer is inaccessible due to 'private' protection level
    static let example: Example = .init()
                                   ^~~~
./example.swift:2:13: note: 'init()' declared here
    private init() {}
```

This behavior does not happen if instead the above line is changed to:

```swift
static let example = Example()
```

**Comment by nicklockwood** [comment]
@msanders thanks for reporting. I'll fix it asap.

**Comment by nicklockwood** [comment]
@msanders fixed in 0.44.1.
